merlin/models/tf/models/base.py

```diff
@@ -736,6 +736,15 @@ def train_step(self, data):
         with tf.GradientTape() as tape:
             x, y, sample_weight = unpack_x_y_sample_weight(data)
 
+            # Ensure that we don't have any ragged or sparse tensors passed at training time.
+            if isinstance(x, dict):
+                for k in x:
+                    if isinstance(x[k], (tf.RaggedTensor, tf.SparseTensor)):
+                        raise ValueError(
+                            "Training with RaggedTensor or SparseTensor inputs is unsupported. "
+                            "Please update your loader to pass dense tensors. "
```

Comment:
```suggestion
                            "Please update your loader to pass dense tensors so that you can save the model, "
                            "and serve it correctly".
```

Reply:
added something along these lines, but needed to split across more lines due to line length constraints